src/chromo/common.py

```diff
@@ -543,6 +537,79 @@ def __getnewargs__(self):
         # upon unpickling, create EventData object instead of MCEvent object
         return (EventData,)
 
+    @abstractmethod
+    def _add_init_beam_info(self):
+        pass
+
+    def _get_ibeam(self):
+        if self.kin._initial_beam_data is None:
+            ibeam = self.kin._get_beam_data(self.kin)
+        else:
+            ibeam = self.kin._initial_beam_data
+
+        if ibeam.ref_frame != self._generator_frame:
+            # Boost current ref frame (ibeam.ref_frame)
+            # to generators ref frame (self._generator_frame)
+            # Use apply_boost use self.kin.frame
+            self.kin.frame = self._generator_frame
+            self.kin.apply_boost(ibeam, ibeam.ref_frame)
+            self.kin.frame = ibeam.ref_frame
+            ibeam.ref_frame = self._generator_frame
+        return ibeam
+
+    def _fill_initial_beam(self):
+        ibeam = self._get_ibeam()
+        for attr_field in ibeam._data_attr:
+            attr_beam = getattr(ibeam, attr_field)
+            attr_event = getattr(self, attr_field)
+            if attr_event is not None:
+                attr_event[0:2] = attr_beam
+                if attr_field in ["mothers"]:
+                    attr_event[
+                        np.isin(attr_event, [[-1, -1], [0, -1], [1, -1], [1, 1]]).all(
+                            axis=1
+                        )
+                    ] = [0, 1]
```

Comment:
What does this do? Please add a comment.

Reply:
It attaches all particles which have beam particles as parents to a single vertex.